homeassistant/components/incomfort/coordinator.py

```diff
@@ -0,0 +1,98 @@
+"""Datacoordinator for InComfort integration."""
+
+from dataclasses import dataclass, field
+from datetime import timedelta
+import logging
+from typing import Any
+
+from aiohttp import ClientResponseError
+from incomfortclient import (
+    Gateway as InComfortGateway,
+    Heater as InComfortHeater,
+    IncomfortError,
+)
+
+from homeassistant.const import CONF_HOST
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+    UpdateFailed,
+)
+from homeassistant.util.hass_dict import HassKey
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+UPDATE_INTERVAL = 30
+
+
+@dataclass
+class InComfortData:
+    """Keep the Intergas InComfort entry data."""
+
+    client: InComfortGateway
+    heaters: list[InComfortHeater] = field(default_factory=list)
+
+
+async def async_connect_gateway(
+    hass: HomeAssistant,
+    entry_data: dict[str, Any],
+) -> InComfortData:
+    """Validate the configuration."""
+    credentials = dict(entry_data)
+    hostname = credentials.pop(CONF_HOST)
+
+    client = InComfortGateway(
+        hostname, **credentials, session=async_get_clientsession(hass)
+    )
+    heaters = await client.heaters()
+
+    return InComfortData(client=client, heaters=heaters)
```

Comment:
If were going to move this, i think init is a better place as this code is purely about initializing the client

Reply:
It is also used in `config_flow.py` and I want to avoid circular import issues